src/gainmap.c

```diff
@@ -704,17 +704,20 @@ avifResult avifRGBImageComputeGainMap(const avifRGBImage * baseRgbImage,
 
     // Scale the gain map values to map [min, max] range to [0, 1].
     for (int c = 0; c < numGainMapChannels; ++c) {
-        const float range = gainMapMaxLog2[c] - gainMapMinLog2[c];
-        if (range <= 0.0f) {
-            continue;
-        }
+        const float range = AVIF_MAX(gainMapMaxLog2[c] - gainMapMinLog2[c], 0.0f);
         const float gainMapGamma = avifUnsignedFractionToFloat(gainMap->gainMapGamma[c]);
 
         for (int j = 0; j < height; ++j) {
             for (int i = 0; i < width; ++i) {
                 // Remap [min; max] range to [0; 1]
-                const float v = AVIF_CLAMP(gainMapF[c][j * width + i], gainMapMinLog2[c], gainMapMaxLog2[c]);
-                gainMapF[c][j * width + i] = powf((v - gainMapMinLog2[c]) / range, gainMapGamma);
+                if (range == 0.0f) {
+                    // If the range is 0, the gain map values will be multiplied by zero when tonemapping so the values
+                    // don't matter, but we still need to make sure that gainMapF is in [0,1].
+                    gainMapF[c][j * width + i] = 0.0f;
+                } else {
+                    const float v = AVIF_CLAMP(gainMapF[c][j * width + i], gainMapMinLog2[c], gainMapMaxLog2[c]);
+                    gainMapF[c][j * width + i] = AVIF_CLAMP(powf((v - gainMapMinLog2[c]) / range, gainMapGamma), 0.0f, 1.0f);
```

Comment:
Since `AVIF_CLAMP` is a macro, the expression `powf((v - gainMapMinLog2[c]) / range, gainMapGamma)` will be evaluated multiple times unless the compiler eliminates the common subexpressions. It would be good to store the result of `powf((v - gainMapMinLog2[c]) / range, gainMapGamma)` in a local variable so that we don't rely on this compiler optimization.

Note: The `AVIF_MAX(gainMapMaxLog2[c] - gainMapMinLog2[c], 0.0f)` expression at line 707 has the same issue, but it is less serious.